### ISSUE TITLE
[Bug]: Sweep with Bayes search fails with "Need at least one searchable parameter" error

### ISSUE THREAD

**Issue by bolak92** [wandb/wandb#8290]
### Issue Description:
I am encountering an issue when attempting to run a sweep using the Bayes search strategy. The error message suggests that no searchable parameter is defined, although the configuration includes both the metric and the parameter search space. Here's the error and the steps leading to it:
The following is the error message
> 400 response executing GraphQL.
{"errors":[{"message":"Invalid sweep config: Need at least one searchable parameter for bayes search.","path":["upsertSweep"]}],"data":{"upsertSweep":null}}
wandb: ERROR Error while calling W&B API: Invalid sweep config: Need at least one searchable parameter for bayes search. (<Response [400]>)
wandb: ERROR Invalid sweep config: Need at least one searchable parameter for bayes search.
Traceback (most recent call last):
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/sdk/lib/retry.py", line 131, in __call__
    result = self._call_fn(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/sdk/internal/internal_api.py", line 345, in execute
    return self.client.execute(*args, **kwargs)  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/vendor/gql-0.2.0/wandb_gql/client.py", line 52, in execute
    result = self._get_result(document, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/vendor/gql-0.2.0/wandb_gql/client.py", line 60, in _get_result
    return self.transport.execute(document, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/sdk/lib/gql_request.py", line 59, in execute
    request.raise_for_status()
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://api.wandb.ai/graphql
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/users/home/bkhalil/Repos/uqdd/uqdd/models/model_parser.py", line 258, in <module>
    query_dict[f"{args.model}_hyperparam"](**kwargs)
  File "/home/bkhalil/Repos/uqdd/uqdd/models/evidential.py", line 228, in run_evidential_hyperparam
    sweep_id = wandb.sweep(sweep_config, project=wandb_project_name)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/sdk/wandb_sweep.py", line 86, in sweep
    sweep_id, warnings = api.upsert_sweep(sweep, prior_runs=prior_runs)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/apis/internal.py", line 133, in upsert_sweep
    return self.api.upsert_sweep(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/apis/normalize.py", line 73, in wrapper
    raise err
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/apis/normalize.py", line 41, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/sdk/internal/internal_api.py", line 3216, in upsert_sweep
    raise e
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/sdk/internal/internal_api.py", line 3210, in upsert_sweep
    response = self.gql(
               ^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/sdk/internal/internal_api.py", line 317, in gql
    ret = self._retry_gql(
          ^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/sdk/lib/retry.py", line 147, in __call__
    retry_timedelta_triggered = check_retry_fn(e)
                                ^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/util.py", line 881, in no_retry_4xx
    raise UsageError(body["errors"][0]["message"])
wandb.errors.UsageError: Invalid sweep config: Need at least one searchable parameter for bayes search.
Traceback (most recent call last):
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/sdk/lib/retry.py", line 131, in __call__
    result = self._call_fn(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/sdk/internal/internal_api.py", line 345, in execute
    return self.client.execute(*args, **kwargs)  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/vendor/gql-0.2.0/wandb_gql/client.py", line 52, in execute
    result = self._get_result(document, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/vendor/gql-0.2.0/wandb_gql/client.py", line 60, in _get_result
    return self.transport.execute(document, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/sdk/lib/gql_request.py", line 59, in execute
    request.raise_for_status()
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://api.wandb.ai/graphql
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/users/home/bkhalil/Repos/uqdd/uqdd/models/model_parser.py", line 258, in <module>
    query_dict[f"{args.model}_hyperparam"](**kwargs)
  File "/home/bkhalil/Repos/uqdd/uqdd/models/evidential.py", line 228, in run_evidential_hyperparam
    sweep_id = wandb.sweep(sweep_config, project=wandb_project_name)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/sdk/wandb_sweep.py", line 86, in sweep
    sweep_id, warnings = api.upsert_sweep(sweep, prior_runs=prior_runs)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/apis/internal.py", line 133, in upsert_sweep
    return self.api.upsert_sweep(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/apis/normalize.py", line 73, in wrapper
    raise err
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/apis/normalize.py", line 41, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/sdk/internal/internal_api.py", line 3216, in upsert_sweep
    raise e
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/sdk/internal/internal_api.py", line 3210, in upsert_sweep
    response = self.gql(
               ^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/sdk/internal/internal_api.py", line 317, in gql
    ret = self._retry_gql(
          ^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/sdk/lib/retry.py", line 147, in __call__
    retry_timedelta_triggered = check_retry_fn(e)
                                ^^^^^^^^^^^^^^^^^
  File "/home/bkhalil/miniconda3/envs/uqdd-118/lib/python3.11/site-packages/wandb/util.py", line 881, in no_retry_4xx
    raise UsageError(body["errors"][0]["message"])
wandb.errors.UsageError: Invalid sweep config: Need at least one searchable parameter for bayes search.
2024-09-06 13:19:42,869 - wandb.wandb_agent - INFO - Cleaning up finished run: p8xnkfd5
`

### What I tried:
* Verified that the metric is correctly specified in the sweep configuration.
* Ensured that the configuration includes valid parameters for Bayes optimization.
* Confirmed during debugging that the metric is part of the sweep_config.

### reproduction steps:
The error is triggered when running the following command for a previously started sweep:
`wandb agent -e bolak92 -p $wandb_project --count $run_count $sweep_id`

The sweep config is a json that I load during the python run and it looks like this:
`{
"program": "model_parser.py",
  "entity": "bolak92",
  "method": "bayes",
  "metric": {
    "goal": "minimize",
    "name": "val/loss"
  },
  "parameters": {...}
  }`
  
  ### Environment:
* wandb version: 0.17.9
* Python version: 3.11.8
* OS: Linux

**Comment by fmamberti-wandb** [comment]
Hi @bolak92, thank you for reaching out and sorry to hear you are experiencing this issue.

Would you be able to share some additional information to help us investigate this:

- the content of the `parameters": {...}` dictionary
- the URL for the sweep in question
- What version of the `wandb` SDK you are on

Thanks!

**Comment by bolak92** [comment]
Hi Thanks for the prompt response. 
* The parameters dictionary is as follows:

> {
  "program": "model_parser.py",
  "entity": "bolak92",
  "method": "bayes",
  "metric": {
    "goal": "minimize",
    "name": "val/loss"
  },
  "parameters": {
    "sweep": {
      "value": true
    },
    "model": {
      "value": "evidential"
    },
    "lamb": {
      "values": [1.0,0.1,0.01,0.001]
    },
    "aleatoric": {
      "value": true
    },
    "chem_layers": {
      "values": [
        [1024,512,256],
        [1024,512],
        [1024,256],
        [1024],
        [512],
        [256]
      ]
    },
    "prot_layers": {
      "values": [
        [1536, 1024, 512, 256],
        [1024,512,256],
        [512,256,128],
        [512,128],
        [512,256],
        [512],
        [256]
      ]
    },
    "regressor_layers": {
      "values": [
        [256,128],
        [128,64],
        [256],
        [128],
        [64]
      ]
    },
    "dropout": {
      "values": [0.1,0.2]
    },
    "batch_size": {
      "values": [128,256,512]
    },
    "epochs": {
      "value": 3000
    },
    "early_stop": {
      "value": 100
    },
    "loss": {
      "value": "evidential_regression"
    },
    "loss_reduction": {
      "value": "mean"
    },
    "optimizer": {
      "values": ["adagrad","adamw","adam","sgd"]
    },
    "lr": {
      "values": [1e-2,1e-3,5e-4]
    },
    "weight_decay": {
      "values": [0,1e-2,1e-4]
    },
    "lr_scheduler": {
      "value": null
    },
    "max_norm": {
      "values": [null,50.0,100.0]
    }
  }
}

* The URL is
https://wandb.ai/bolak92/evidential/sweeps/asgvot1p/overview

* My wandb version is 0.17.9

**Comment by bolak92** [comment]
I solved the problem.

For the record, the issue was in my own script. 
when I was calling the agent again with `wandb agent` it was keeping an argument from the config `--sweep_count` which I have defined as the condition in the script whether to run hyperparameter optimization vs normal model run. because it wasn't popped out of the kwargs dictionary that I was feeding to the config dictionary definition, it was kept with the downstream runs as well.

that being said, it was trying to rerun the sweep hpo script on top of the previous sweep hpo script.

Moving forward, I think a better error message can be very useful in pointing better to this specific issue. 
I realized the problem by looking quite close into the commands that are being run through the script with `wandb agent`
